pkg/sql/schemaexpr/expr.go

```diff
@@ -136,6 +136,21 @@ func FormatExprForDisplayWithoutTypeAnnotations(
 	return tree.AsString(expr), nil
 }
 
+// FormatExprForDisplayInPGCatalog formats a schema expression string for
+// display, similar to FormatExprForDisplay, but does not add type annotations
+// and does add typecasts for non-numeric constants. This is the format that
+// should be used for displaying expressions in pg_catalog, in order to achieve
+// compatibility with the pg_catalog from PostgreSQL.
+func FormatExprForDisplayInPGCatalog(
```

Comment:
[nit] Create a format option enum and combine these three `FormatExprForDisplay...` functions into one. They're all very similar.

Reply:
good call